src/crimson/os/seastore/cache.h

```diff
@@ -253,11 +264,14 @@ class Cache {
     CachedExtentRef ret;
     LOG_PREFIX(Cache::get_extent_if_cached);
     auto result = t.get_extent(offset, &ret);
-    if (result != Transaction::get_extent_ret::ABSENT) {
-      // including get_extent_ret::RETIRED
-      SUBDEBUGT(seastore_cache,
-	"Found extent at offset {} on transaction: {}",
-	t, offset, *ret);
+    if (result == Transaction::get_extent_ret::RETIRED) {
```

Comment:
should we return empty CachedExtentRef here? IIUC `TransactionManager::get_extent_if_live` and `SegmentCleaner::gc_reclaim_space()` should not rewrite retired extents.